internal/fuse/file.go

```diff
@@ -111,6 +90,27 @@ func (f *file) Read(ctx context.Context, req *fuse.ReadRequest, resp *fuse.ReadR
 	debug.Log("Read(%v, %v, %v), file size %v", f.node.Name, req.Size, req.Offset, f.node.Size)
 	offset := req.Offset
 
+	if !f.hasSizes {
```

Comment:
Shouldn't this be protected by a lock? Or does FUSE handle that?

Reply:
Never mind. FUSE calls Lookup, and Lookup returns a fresh *file every time. So this is safe.

Reply:
Thanks for researching! I wasn't sure about how FUSE handles concurrency, but `getBlobAt` also seems not to be concurrency-safe, so I assumed it would be handled by FUSE...

Reply:
Actually, I failed to spot that the bazil.org/fuse package stores the node it gets from Lookup in a cache, and it's concurrent. So there really should be locks all over internal/fuse.

Reply:
False alarm. It looks like the kernel driver will synchronize access, as long as you don't set the AsyncRead MountOption. I've empirically verified this by setting minSnapshotsReloadTime to 0, running restic mount with the race detector on, reading the snapshots directory concurrently in two loops and adding snapshots.

Reply:
The documentation of bazil/fuse actually says that synchronization is required (see https://godoc.org/bazil.org/fuse#hdr-Service_Methods):

> Multiple goroutines may call service methods simultaneously; the methods being called are responsible for appropriate synchronization.

I'd assume that some caching in the kernel effectively synchronizes access, but it probably not a good idea to rely on it.

Reply:
I added a Mutex to synchronize access